src/main/java/seedu/jarvis/model/greeting/Greeting.java

```diff
@@ -24,14 +24,19 @@ public StringProperty getGreeting() {
     }
 
     public void setGreeting(String newGreeting) {
-        if (newGreeting == null || newGreeting.isEmpty() || newGreeting.equals(DEFAULT_GREETING)) {
+        if (verifyGreeting(newGreeting)) {
             this.greeting.setValue(DEFAULT_GREETING);
         } else {
             this.greeting.setValue(padWithIntro(newGreeting));
         }
 
     }
 
+    private boolean verifyGreeting(String greeting) {
+        return greeting == null || greeting.isEmpty() || greeting.equals(DEFAULT_GREETING)
+                || greeting.contains("Welcome");
```

Comment:
Does this mean that the when the user switches to a non-TA account, the greeting will remain as the previous TA's name?

Reply:
yes, good too cos if ppl change user, the greeting should change as well.